sdk/src/main/java/com/deploygate/sdk/DeployGate.java

```diff
@@ -892,20 +896,50 @@ static DeployGate getInstance() {
         return sInstance;
     }
 
-    void sendCrashReport(Throwable ex) {
+    void sendCrashReport(/* non-null */ Throwable ex) {
         if (mRemoteService == null)
             return;
         
         Bundle extras = new Bundle();
-        extras.putSerializable(DeployGateEvent.EXTRA_EXCEPTION, ex);
         try {
+            if (getDeployGateVersionCode() > 41 /* greater than ver 1.7.2 */) {
+                Throwable rootCause = getRootCause(ex);
+                String msg = rootCause.getMessage();
+                extras.putString(DeployGateEvent.EXTRA_EXCEPTION_ROOT_CAUSE_CLASSNAME, rootCause.getClass().getName());
+                extras.putString(DeployGateEvent.EXTRA_EXCEPTION_ROOT_CAUSE_MESSAGE, msg != null ? msg : "");
+                extras.putString(DeployGateEvent.EXTRA_EXCEPTION_STACKTRACES, Log.getStackTraceString(ex));
+            } else {
+                extras.putSerializable(DeployGateEvent.EXTRA_EXCEPTION, ex);
+            }
+
             mRemoteService.sendEvent(mApplicationContext.getPackageName(),
                     DeployGateEvent.ACTION_SEND_CRASH_REPORT, extras);
         } catch (RemoteException e) {
             Log.w(TAG, "failed to send crash report: " + e.getMessage());
         }
     }
 
+    /* non-null */
+    private static Throwable getRootCause(/* non-null */ Throwable ex) {
+        Throwable cause = ex;
+
+        List<Throwable> throwables = new ArrayList<>(Collections.singleton(cause));
+
+        Throwable rootCause = cause;
+
+        while (true) {
```

Comment:
The execution path of this loop looks a bit complicated. I'd prefer simplifying this method something like:

```java
Throwable cause = ex;
LinkedList<Throwable> throwables = new LinkedList<>();
while (cause != null && !throwables.contains(cause)) {
    throwables.add(cause);
    cause = cause.getCause();
}
return throwables.getLast();
```

Reply:
Addressed in https://github.com/DeployGate/deploygate-android-sdk/pull/14/commits/5c25059d2114aeabfb616595ac94a5e345622e11